src/vars.rs

```diff
@@ -61,6 +82,9 @@ impl Vars {
     pub fn get(&self, k: impl AsRef<str>) -> &Var {
         let k = k.as_ref().trim();
         match k {
+            // Special variables.
+            "MAKE" => &self.make_var,
+            "MAKEFLAGS" => &self.makeargs_var,
```

Comment:
I believe both of these variables are allowed to be modified inside the makefile, so we probably shouldn't treat them differently.